### PR TITLE
Add doc about migrate tasks to node10

### PR DESCRIPTION
**Pull request by AnnaOpareva**
Add a guideline for migrating tasks to Node 10.

[link to new doc](https://github.com/microsoft/azure-pipelines-tasks/blob/users/v-aopareva/add_doc_about_migrate_tasks_node10/docs/migrateNode10.md)

### CODE REVIEW

File: docs/migrateNode10.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  - [`fs` module](#fs-module)
+
+# Upgrading Tasks to Node 10
+
+1. Upgrade `typescript` to `4.0.2` version and fix type errors. Add the following snippet to the `package.json`:
+```
+  "devDependencies": {
+    "typescript": "^4.0.0"
+  }
+```
```
Comment (alexander-smolyakov) [inline docs/migrateNode10.md:14]:
```suggestion
```json
  "devDependencies": {
    "typescript": "^4.0.0"
  }
``` 
```

File: docs/migrateNode10.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+Use the latest major version of a "common package" at `common-npm-packages` folder, which depends on the `azure-pipelines-task-lib` package with `^3.1.7` version.
+
+The task-lib package uses some shared (e.g. global object) resources to operate so it may cause unexpected errors in cases when more than one version of the package is installed for a task. It happens in the case of a child package's task-lib dependency has a different version than a task's `task-lib` has.
+
+If you are planning to move some common package to common-npm-packages directory - please note that you need to update all necessary paths in this package ([example of such path](https://github.com/microsoft/azure-pipelines-tasks/blob/master/common-npm-packages/packaging-common/Tests/MockHelper.ts#L44))
+
+# List of known dependency issues
+
+### fs module
```
Comment (alexander-smolyakov) [inline docs/migrateNode10.md:81]:
```suggestion
## `fs` module
````

File: docs/migrateNode10.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+5. Upgrade any additional dependencies that may be incompatible with Node 10.
+An example is the `sync-request` package, which needs to be upgraded to the latest version from v3.0.1
+See some additional [dependency issues](#list-of-known-dependency-issues) below.
+
+6. Thoroughly test tasks with unit tests and on an actual agent. The build agent now supports Node 10, so testing can be done on live versions of Azure DevOps.
+
+7. Bumping the minimum agent version is not required, as the server will enforce a minimum version for pipelines containing Node 10 tasks.
+
+## Common packages dependent on azure-pipeline-task-lib
```
Comment (alexander-smolyakov) [inline docs/migrateNode10.md:71]:
```suggestion
## Common packages dependent on `azure-pipeline-task-lib`
```

**Review by alexander-smolyakov — APPROVED** [review]
LGTM and please take a look at the suggestions below.

**Review by anatolybolshakov — APPROVED** [review]
LGTM!

**Review by EzzhevNikita — APPROVED** [review]

**Pull request merged**